### PR TITLE
getStatistic method improved

### PR DESCRIPTION
**Pull request by ivanlavrinenko**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
+        int result = sumOfSupplies - sumOfBuying;
+        StringBuilder report = new StringBuilder();
+        report.append("supply,").append(sumOfSupplies).append(System.lineSeparator())
+                .append("buy,").append(sumOfBuying).append(System.lineSeparator())
+                .append("result,").append(result);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            writer.write(report.toString());
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to the file " + toFileName, e);
+        }
```
Comment (Sanderfuck) [inline src/main/java/core/basesyntax/WorkWithFile.java:46]: image

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+            } else {
+                String buy = row.split(",")[AMOUNT_INDEX];
+                sumOfBuying += Integer.parseInt(buy);
+            }
+        }
+        int result = sumOfSupplies - sumOfBuying;
+        StringBuilder report = new StringBuilder();
+        report.append("supply,").append(sumOfSupplies).append(System.lineSeparator())
+                .append("buy,").append(sumOfBuying).append(System.lineSeparator())
+                .append("result,").append(result);
```
Comment (Sanderfuck) [inline src/main/java/core/basesyntax/WorkWithFile.java:41]: image

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,7 +1,48 @@
...
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String value = reader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from the file " + fromFileName);
+        }
+        String [] dataFromFileName = stringBuilder.toString().split(System.lineSeparator());
```
Comment (Sanderfuck) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]: Divide this line

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by ivanlavrinenko — COMMENTED** [review]
methods readDataFromFile, generateReport, writeReportToFile were created, used string format to create report

**Review by sokimaaa — APPROVED** [review]

**Pull request closed without merging**